test-alerts.cpp

```diff
@@ -6,3 +6,74 @@
 TEST_CASE("infers the breach according to limits") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
 }
+
+TEST_CASE("Check temperature classification") {
+  std::unordered_map <int ,std::pair<int, int>> testTempClassify;
+  classifyTemperatureBreach(testTempClassify);
+  
+  auto itr = testTempClassify.find(PASSIVE_COOLING);
+  REQUIRE((itr->second.first) == 0);
+  REQUIRE((itr->second.second) == 35);
+
+  itr = testTempClassify.find(HI_ACTIVE_COOLING);
+  REQUIRE((itr->second.first) == 0);
+  REQUIRE((itr->second.second) == 45);
+
+  itr = testTempClassify.find(MED_ACTIVE_COOLING);
+  REQUIRE((itr->second.first) == 0);
+  REQUIRE((itr->second.second) == 40);
+}
+
+TEST_CASE("Send alert") {
+  std::ostringstream lowToEmail;
+  sendBatteryTempStatus(TO_EMAIL, TOO_LOW, lowToEmail, "krish@gmail.com");
+  REQUIRE(lowToEmail.str() == "To: krish@gmail.com\nHi, the temperature is too low\n");
+
+  std::ostringstream highToEmail;
+  sendBatteryTempStatus(TO_EMAIL, TOO_HIGH, highToEmail, "krish@gmail.com");
+  REQUIRE(highToEmail.str() == "To: krish@gmail.com\nHi, the temperature is too high\n");
+
+  std::ostringstream lowToController;
+  sendBatteryTempStatus(TO_CONTROLLER, TOO_LOW, lowToController);
+  REQUIRE(lowToController.str() == "65261 : 1\n");
+
+  std::ostringstream highToController;
+  sendBatteryTempStatus(TO_CONTROLLER, TOO_HIGH, highToController);
+  REQUIRE(highToController.str() == "65261 : 2\n");
+}
+
+TEST_CASE("Test checkAndAlert(to email)") {
+
+  BatteryCharacter batterych = {
+    .coolingType = HI_ACTIVE_COOLING
+  };
+
+  std::ostringstream toEmail;
+  std::streambuf* streambuf_1 = std::cout.rdbuf();
+  std::cout.rdbuf(toEmail.rdbuf());
+
+  checkAndAlert(TO_EMAIL, batterych, 20);
```

Comment:
how do we verify if the intended message has been received by the right target, and right recipient
Test should be present for that too